src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -170,9 +182,10 @@ public void ScheduleForExecution(AsyncTask asyncTask)
         }
 
         /// <summary>
-        ///     Flag determining whether or not the scheduler is operating in Test Mode.
+        /// Flag determining whether or not the scheduleed task will be
+        /// processed immediately or not. 
         /// </summary>
-        public bool IsTestMode { get; private set; }
+        public TaskProcessMode ProcessMode { get; set; }
```

Comment:
`IsTestMode` is used also to determine if a given operation needs to be done, for example, logging for instrumentation is turned off during test runs. I agree that Dynamo code should behave the same for both real-life and test scenarios, and this flag should be avoided at all times (so our tests actually test what the end users will use in the end).